velox/vector/tests/DecodedVectorTest.cpp

```diff
@@ -289,15 +289,42 @@ class DecodedVectorTest : public testing::Test, public VectorTestBase {
       }
     };
 
+    // Test dict-over-constant encoding structure on a complex type.
+    auto checkDepthOne = [&](auto& decoded, auto& wrappedVector) {
```

Comment:
Also, I find checkDepthOne name confusing. Any particular reason the testing logic can't be folded into 'check' lambda? I assume the logic here is to wrap the base of the decoded vector using DecodedVector::wrap and verify that the resulting vector equals the original vector. Is this so?

Reply:
Good suggestion.
One caveat: once merging the check lambdas, the dict(null constant) check will be skipped due to a potential bug: https://github.com/facebookincubator/velox/issues/3704